server/src/bin/ingestion-worker.rs

```diff
@@ -333,25 +321,233 @@ async fn ingestion_worker(
+                convert_html_to_text(&message.chunk.chunk_html.clone().unwrap_or_default())
+            } else {
+                message.chunk.chunk_html.clone().unwrap_or_default()
+            };
+
+            let qdrant_point_id = message
+                .ingest_specific_chunk_metadata
+                .qdrant_point_id
+                .unwrap_or(uuid::Uuid::new_v4());
+
+            let chunk_tag_set = message
+                .chunk
+                .tag_set
+                .clone()
+                .map(|tag_set| tag_set.join(","));
+
+            let timestamp = {
+                message
+                    .chunk
+                    .time_stamp
+                    .clone()
+                    .and_then(|ts| -> Option<NaiveDateTime> {
+                        ts.parse::<DateTimeUtc>()
+                            .ok()
+                            .map(|date| date.0.with_timezone(&chrono::Local).naive_local())
+                    })
+            };
+
+            let chunk_tracking_id = message
+                .chunk
+                .tracking_id
+                .clone()
+                .filter(|chunk_tracking| !chunk_tracking.is_empty());
+
+            let chunk_metadata = ChunkMetadata {
+                id: message.ingest_specific_chunk_metadata.id,
+                content: content.clone(),
+                link: message.chunk.link.clone(),
+                qdrant_point_id: Some(qdrant_point_id),
+                created_at: chrono::Utc::now().naive_local(),
+                updated_at: chrono::Utc::now().naive_local(),
+                tag_set: chunk_tag_set,
+                chunk_html: message.chunk.chunk_html.clone(),
+                metadata: message.chunk.metadata.clone(),
+                tracking_id: chunk_tracking_id,
+                time_stamp: timestamp,
+                dataset_id: payload.dataset_id,
+                weight: message.chunk.weight.unwrap_or(0.0),
+            };
+
+            (
+                chunk_metadata,
+                message.chunk.file_id,
+                message.chunk.group_ids.clone(),
+                message.upsert_by_tracking_id,
+            )
+        })
+        .collect();
+
+    let inserted_chunk_metadatas =
+        bulk_insert_chunk_metadata_query(ingestion_data, payload.dataset_id, web_pool.clone())
+            .await?;
+
+    if inserted_chunk_metadatas.is_empty() {
+        // All collisions
+        return Ok(vec![]);
+    }
+
+    // Only embed the things we get returned from here, this reduces the number of times we embed
+    // data that are just duplicates
+    let all_content: Vec<String> = inserted_chunk_metadatas
+        .iter()
+        .map(|(chunk_metadata, _, _, _)| {
+            // handle
+            convert_html_to_text(
+                &chunk_metadata
+                    .chunk_html
+                    .clone()
+                    .unwrap_or(chunk_metadata.content.clone()),
+            )
```

Comment:
Cant we just the content field rather than converting to html again?

Reply:
#Yes we Can. Updating